src/components/QuestionDetails/index.js

```diff
@@ -0,0 +1,144 @@
+import {useState} from 'react';
+import {useNavigate} from 'react-router-dom';
+import {useParams} from 'react-router-dom';
+
+import {useStore} from '../../hooks/useStore';
+import AnswerCard from '../AnswerCard';
+import Button from '../Button';
+import Form from '../Form';
+
+import {StyledQuestionDetails} from './styled';
+import {StyledNavigation} from './styled';
+import {StyledQuestionBody} from './styled';
+import {StyledQuestionHead} from './styled';
+import {StyledImage} from './styled';
+
+export default function QuestionDetails() {
+	const [showEditQuestionForm, setShowEditQuestionForm] = useState(false);
+	const questions = useStore(state => state.questions.data);
+	const answers = useStore(state => state.answers.data);
+	const setQuestion = useStore(state => state.setQuestion);
+	const addNewAnswer = useStore(state => state.addNewAnswer);
+	const activeUser = useStore(state => state.activeUser);
+
+	const fetchData = useStore(state => state.fetchData);
+	const navigate = useNavigate();
+	const {idFromUrl} = useParams();
+
+	const question = questions.find(question => question._id === idFromUrl);
+
+	const addAnswer = event => {
+		event.preventDefault();
+		const newAnswer = {
+			answerText: event.target.inputField.value,
+			questionID: question._id,
+		};
+		addNewAnswer(newAnswer);
+		event.target.reset();
+	};
+
+	const editQuestion = event => {
+		event.preventDefault();
+		const newQuestion = question;
+		newQuestion.questionText = event.target.inputField.value;
+		setQuestion(question._id, newQuestion);
+		fetchData();
+		event.target.reset();
+	};
+	const closeQuestion = () => {
+		const newQuestion = question;
+		newQuestion.answered = !question.answered;
+		setQuestion(question._id, newQuestion);
+		fetchData();
+	};
+
+	return (
+		<StyledQuestionDetails>
+			<StyledNavigation>
+				<Button
+					onClick={() => {
+						navigate(-1);
+					}}
+				>
+					<svg width="2em" height="2em" viewBox="0 0 24 24">
+						<path
+							fill="currentColor"
+							d="M2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12M18,11H10L13.5,7.5L12.08,6.08L6.16,12L12.08,17.92L13.5,16.5L10,13H18V11Z"
+						/>
+					</svg>
+				</Button>
+				<StyledImage
+					id="statusIndicator"
```

Comment:
What are you doing with this, a kind of weird id? 🤔

Reply:
i`ve replaced it with an styled component :)